### PR TITLE
Add `ViT` example

### PR DESCRIPTION
**Pull request by jafermarq**
This example shows how to federate the finetuning of a ViT which is pretrained on ImageNet. The finetuning dataset is CIFAR-100, which is partitioned into 20 clients using `flower-datasets`. 

The example runs with both `start_simulation()` and Flower Next's `flower-simulation`

<examples>

### CODE REVIEW

File: examples/vit-finetune/README.md [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+
+## Running the example
+
+If you haven't cloned the Flower repository already you might want to clone code example and discard the rest. We prepared a single-line command that you can copy into your shell which will checkout the example for you:
+
+```shell
+git clone --depth=1 https://github.com/adap/flower.git && mv flower/examples/vit-finetune . && rm -rf flower && cd vit-finetune
+```
+
+This will create a new directory called `simulation-pytorch` containing the following files:
```
Comment (danieljanes) [inline examples/vit-finetune/README.md:13]:
```suggestion
This will create a new directory called `vit-finetune` containing the following files:
```

File: examples/vit-finetune/README.md [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+
+```
+-- README.md         <- Your're reading this right now
+-- main.py           <- Main file that launches the simulation
+-- client.py         <- Contains Flower client code and ClientApp
+-- server.py         <- Contains Flower server code and ServerApp
+-- model.py          <- Defines model and train/eval functions
+-- dataset.py        <- Downloads, partitions and processes dataset
+-- pyproject.toml    <- Example dependencies
+-- requirements.txt  <- Example dependencies
```
Comment (danieljanes) [inline examples/vit-finetune/README.md:23]:
```suggestion
-- pyproject.toml    <- Example dependencies, installable using Poetry
-- requirements.txt  <- Example dependencies, installable using pip
```

File: examples/vit-finetune/README.md [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+|    0   N/A  N/A     31421      C   ray::DefaultActor.run                      1536MiB |
+|    0   N/A  N/A     31422      C   ray::DefaultActor.run                      1536MiB |
++---------------------------------------------------------------------------------------+
+```
+
+### Run with Flower Next (preview)
+
+```bash
+flower-simulation --client-app=client:app --server-app=server:app --num-supernodes=20 \
+    --backend-config='{"client_resources": {"num_cpus":4, "num_gpus":0.25}}'
```
Comment (danieljanes) [inline examples/vit-finetune/README.md:90]:
```suggestion
flower-simulation \
    --client-app=client:app \
    --server-app=server:app \
    --num-supernodes=20 \
    --backend-config='{"client_resources": {"num_cpus":4, "num_gpus":0.25}}'
```

File: examples/vit-finetune/client.py [hunk 4]
```diff
@@ -0,0 +1,84 @@
+import torch
+from torch.utils.data import DataLoader
+
+import flwr
+from flwr.client import NumPyClient
+
+
```
Comment (danieljanes) [inline examples/vit-finetune/client.py:7]: This example doesn't seem to be auto-formatted, can we run `format` once?
```suggestion

```

File: examples/vit-finetune/pyproject.toml [hunk 5]
```diff
@@ -0,0 +1,17 @@
+[build-system]
+requires = ["poetry-core>=1.4.0"]
+build-backend = "poetry.core.masonry.api"
+
+[tool.poetry]
+name = "quickstart-pytorch"
```
Comment (danieljanes) [inline examples/vit-finetune/pyproject.toml:6]:
```suggestion
name = "vit-finetune"
```

File: examples/vit-finetune/model.py [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+            optimizer.zero_grad()
+            loss = criterion(net(images), labels)
+            avg_loss += loss.item() / labels.shape[0]
+            loss.backward()
+            optimizer.step()
+
+    return avg_loss / len(trainloader)
+
+
+# borrowed from Pytorch quickstart example
```
Comment (danieljanes) [inline examples/vit-finetune/model.py:58]:
```suggestion
```

**Review by danieljanes — DISMISSED** [review]

**Review by danieljanes — APPROVED** [review]
